File: plugins/node/template/addons/email/src/config/email.ts
import { z } from "zod";

const emailSchema = z.object({
  MAIL_HOST: z.string().min(1, "MAIL_HOST is required"),
  MAIL_PORT: z.coerce.number().int().default(587),
  MAIL_USERNAME: z.string().min(1, "MAIL_USERNAME is required"),
  MAIL_PASSWORD: z.string().min(1, "MAIL_PASSWORD is required"),
  MAIL_FROM_ADDRESS: z.string().email("MAIL_FROM_ADDRESS must be a valid email"),
  MAIL_FROM_NAME: z.string().default("{{PROJECT_NAME}}"),
});

export const emailEnv = emailSchema.parse(process.env);

File: plugins/python/template/addons/email/app/core/email.py
from pydantic_settings import BaseSettings, SettingsConfigDict


class EmailSettings(BaseSettings):
    MAIL_HOST: str = "smtp.gmail.com"
    MAIL_PORT: int = 587
    MAIL_USERNAME: str = ""
    MAIL_PASSWORD: str = ""
    MAIL_FROM_ADDRESS: str = ""
    MAIL_FROM_NAME: str = "{{PROJECT_NAME}}"
    MAIL_USE_TLS: bool = True
    MAIL_USE_SSL: bool = False

    model_config = SettingsConfigDict(env_file=".env", case_sensitive=True, extra="ignore")


email_settings = EmailSettings()
